### PR TITLE
Removed all Compiler Warnings when Building KOS

### PR DESCRIPTION
**Pull request by gyrovorbis**
There are a bunch of dumb and mostly harmless warnings in a bunch of files when building KOS, and I think newer GCC versions are starting to get even more whiny about certain things that used to slide. It's been on my agenda to fix them.

- bunch of signed vs unsigned comparisons warnings
- bunch of "X can be used uninitialized" warnings
- several variables declared and never used
- errno setting when erasing a directory in `fs_fat.c` looks like it may have actually been set wrong

Note that my initial work on these warnings is with GCC12, because I think it's the angriest one, but I will make sure they're resolved in the other toolchains as well. 

EDIT: Done. @cepawiel just verified warnings are all gone with legacy and stable toolchains as well. THANK YOU!

### CODE REVIEW

File: addons/libkosfat/directory.h [hunk 1]
```diff
@@ -14,6 +14,8 @@ __BEGIN_DECLS
 
 #include "fatfs.h"
 
+#define PACKED __attribute__((packed))
+
```
Comment (ljsebald) [inline addons/libkosfat/directory.h:18]: Did you mean to change the two instances of `__attribute__((packed))` down below to use this? Otherwise, why is this needed?

File: kernel/libc/koslib/realpath.c [hunk 2]
```diff
@@ -26,11 +26,10 @@
  * SUCH DAMAGE.
  */
 
+#include <kos/limits.h>
```
Comment (ljsebald) [inline kernel/libc/koslib/realpath.c:29]: Why'd this need to be moved?

Comment (andressbarajas) [reply kernel/libc/koslib/realpath.c:29]: Looks like he was going from bigger to smaller depending on the length of the includes.  I do this too.

**Review by ljsebald — APPROVED** [review]
